dusty/src/main.rs

```diff
@@ -22,32 +23,43 @@ use std::fs::File;
 use std::fs::OpenOptions;
 use std::path::Path;
 use std::path::PathBuf;
-use structopt::StructOpt;
 use usdt_impl::Error as UsdtError;
 
 /// Inspect data related to USDT probes in object files.
-#[derive(Debug, StructOpt)]
+#[derive(Debug, Parser)]
 struct Cmd {
     /// The object file to inspect
-    #[structopt(parse(from_os_str))]
     file: PathBuf,
 
     /// Operate more verbosely, printing all available information
-    #[structopt(short, long)]
+    #[arg(short, long)]
     verbose: bool,
 
     /// Print raw binary data along with summaries or headers
-    #[structopt(short, long)]
+    #[arg(short, long, conflicts_with = "json")]
```

Comment:
I think a `--format` argument that accepts the `FormatMode` directly might be more flexible? One would need to add a `FromStr` implementation for `FormatMode`, but that should be it.

Reply:
Perhaps, yeah, I just feel like it's maybe not as terse. IF we did that, we'd want to wrap `FormatMode` in a struct and define that in this module. If you're okay with it I'd just like to move forward with this shape and we can re-evaluate if/when another state needs to be added.